chromadb/execution/expression/operator.py

```diff
@@ -0,0 +1,38 @@
+from dataclasses import dataclass
+from typing import Optional
+from uuid import UUID
+
+from chromadb.types import Where, WhereDocument
+from chromadb.api.types import Embeddings, IDs
+
+
+@dataclass
+class Scan:
+    segment: UUID
+    collection: UUID
+
+
+@dataclass
+class Filter:
+    user_ids: Optional[IDs] = None
+    where: Optional[Where] = None
+    where_document: Optional[WhereDocument] = None
+
+
+@dataclass
+class KNN:
```

Comment:
Does KNN not need a segment?

Reply:
Update: The segment information is managed by `SegmentManager`, and it's easier to let the segment manager to decide which specific version of segment to use. The local segment manager will directly execute the query play based on the collection information and segment information it has, while the distributed segment manager will attach the segment information to the query plan before passing it down to the query node.